File: Data Structures & Algorithms/first-missing-positive/submission-7.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        int i = 0;
        
        while (i < nums.size()) {
            int pos = nums[i] - 1;
            if (nums[i] > 0 && nums[i] <= (int)nums.size() && nums[i] != nums[pos]) swap(nums[i], nums[pos]);
            else i ++;
        }

        i = 0;

        while(i < nums.size()) {
            if (i+1 != nums[i]) break;
            i ++;
        }

        return i + 1;
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-7.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        m = dict()

        for i in range(len(nums)):
            key = target - nums[i]

            if nums[i] in m: return [m[nums[i]], i]
            m[key] = i
        
        return []
